homeassistant/data_entry_flow.py

```diff
@@ -25,6 +25,10 @@
 EVENT_DATA_ENTRY_FLOW_PROGRESSED = "data_entry_flow_progressed"
 
 
+class BaseServiceInfo:
```

Comment:
Should we decorate it with `@dataclass`?

Reply:
I felt it was unecessary since it didn't have any properties, but I can add the decorator.

And since we use a base class, should we bring `__getitem__` in the base class?

Reply:
Note: @bdraco previously recommended that __getitem__ be implement in the child classes because `other discovery types might longer transition periods it might be best to implement this as part of DhcpServiceInfo since it won't be possible to change every discovery flow in a single change.`

Reply:
I have added the decorator (and thanks to #60194 it cancelled the previous CI run)